python/cudf_polars/cudf_polars/dsl/expressions/rolling.py

```diff
@@ -216,17 +253,48 @@ def __init__(
             for ne in self.named_aggs
             for v in (ne.value,)
             if isinstance(v, expr.Agg)
-            or (isinstance(v, expr.UnaryFunction) and v.name == "rank")
+            or (isinstance(v, expr.UnaryFunction) and v.name in {"rank"})
         ]
         self.by_count = len(by_expr)
-        self.children = tuple(by_expr) + tuple(child_deps)
+        self.children = tuple(
+            itertools.chain(
+                by_expr,
+                (() if self._order_by_expr is None else (self._order_by_expr,)),
+                child_deps,
+            )
+        )
+
+    def _sorted_grouper(self, by_cols_for_scan: list[Column]) -> plc.groupby.GroupBy:
+        return plc.groupby.GroupBy(
+            plc.Table([c.obj for c in by_cols_for_scan]),
+            null_handling=plc.types.NullPolicy.INCLUDE,
+            keys_are_sorted=plc.types.Sorted.YES,
+            column_order=[k.order for k in by_cols_for_scan],
+            null_precedence=[k.null_order for k in by_cols_for_scan],
+        )
+
+    @singledispatchmethod
+    def _apply_unary_op(
+        self,
+        op: UnaryOp,
+        _: DataFrame,
+        __: plc.groupby.GroupBy,
+    ) -> tuple[list[str], list[DataType], list[plc.Table]]:
+        raise NotImplementedError(
+            f"Unsupported unary op: {type(op).__name__}"
+        )  # pragma: no cover; translation raises first
 
-    def _rank_group_by_scan(
+    @_apply_unary_op.register
+    def _(
```

Comment:
Note: I chose to write "handler" methods via `singledispatchmethod` because I think it makes it easier to support new unary operations on grouped windows. All one needs to do is add a new `UnaryOp` type and an implementation in `_apply_unary_op`.